serviceconnector/src/createServiceConnector/AuthenticationTypeStep.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+import { AuthInfoBase, KnownAuthType } from "@azure/arm-servicelinker";
+import { AzureWizardPromptStep, IAzureQuickPickItem } from "@microsoft/vscode-azext-utils";
+import * as vscode from 'vscode';
+import { ICreateLinkerContext } from "./ICreateLinkerContext";
+
+export class AuthenticationTypeStep extends AzureWizardPromptStep<ICreateLinkerContext>{
+    public async prompt(context: ICreateLinkerContext): Promise<void> {
+        const authInfoLabels: string[] = [
```

Comment:
Why are these in an array?

I think it'd be better to just put the values directly into the `picks` array below.

Reply:
I think I've seen Matt use this sort of design pattern as well. But I'm also in favor of just putting it in the picks array. It seems like it would be easier to have a mismatch of label/data pairs.

Reply:
Yeah I've done something similar for `ImageSourceListStep`.  My reason for doing so was also readability, because we were conditionally pushing in separate picks across multiple lines, I thought it would be easier to be able to look at the list of labels at the top and quickly get an idea of what all the prompts being used were.  Not super opinionated about it though, would have been fine to add it directly into each pick.

In this case the picks are all constructed together anyway though, so it doesn't really benefit in the same way.